src/components/__tests__/highlight.js

```diff
@@ -50,3 +57,28 @@ test('should render an empty string if attribute is not highlighted', () => {
 
   expect(vm.$el.outerHTML).toMatchSnapshot();
 });
+
+test('should throw an error when not in production if attribute is not highlighted', () => {
+  global.console = { error: jest.fn() };
+
+  const result = {
+    _highlightResult: {},
+  };
+
+  new Vue({
+    template: '<highlight attributeName="attr" :result="result">',
+    render(h) {
+      return h('highlight', {
+        props: {
+          attributeName: 'attr',
+          result,
+        },
+      });
+    },
+    components: {
+      Highlight,
+    },
+  }).$mount();
+
+  expect(global.console.error).toHaveBeenCalled();
```

Comment:
This is because Vue catches errors happening during rendering. See: https://vuejs.org/v2/api/#errorHandler

This PR is not finished yet though, there is some cleanup coming.